WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -875,4 +962,286 @@ private void populateSelectedCategories() {
+                        mFeaturedImageView.setVisibility(View.VISIBLE);
+                        mFeaturedImageViewLocal.setVisibility(View.GONE);
+                        addRemoteMedia(media.getSource());
+                        ++libraryMediaAdded;
+                    } else { // local images
+                        mFeaturedImageView.setVisibility(View.GONE);
+                        mFeaturedImageViewLocal.setVisibility(View.VISIBLE);
+                        mLoadingFeaturedImageIndicator.setVisibility(View.VISIBLE);
+                        addLocalMedia(media.getSource());
+                        ++localMediaAdded;
+                    }
+                }
+
+                if (localMediaAdded > 0) {
+                    Map<String, Object> analyticsProperties = new HashMap<>();
+                    analyticsProperties.put(ANALYTIC_PROP_NUM_LOCAL_PHOTOS_ADDED, localMediaAdded);
+                    AnalyticsTracker.track(AnalyticsTracker.Stat.EDITOR_ADDED_PHOTO_VIA_LOCAL_LIBRARY, analyticsProperties);
+                }
+
+                if (libraryMediaAdded > 0) {
+                    Map<String, Object> analyticsProperties = new HashMap<>();
+                    analyticsProperties.put(ANALYTIC_PROP_NUM_WP_PHOTOS_ADDED, libraryMediaAdded);
+                    AnalyticsTracker.track(AnalyticsTracker.Stat.EDITOR_ADDED_PHOTO_VIA_WP_MEDIA_LIBRARY, analyticsProperties);
+                }
+            } else {
+                mSetFeaturedImageButton.setVisibility(View.VISIBLE);
+                mRemoveFeaturedImageButton.setVisibility(View.GONE);
+            }
+        }
+    }
+
+    /**
+     * Get the maximum size a thumbnail can be to fit in either portrait or landscape orientations.
+     */
+    public int maxFeaturedImageThumbnailWidth() {
+        if (mFeaturedImageThumbnailWidth == 0) {
+            Point size = DisplayUtils.getDisplayPixelSize(getActivity());
+            int screenWidth = size.x;
+            int screenHeight = size.y;
+            mFeaturedImageThumbnailWidth = (screenWidth > screenHeight) ? screenHeight : screenWidth;
+            // 16dp of padding on each side so you can still place the cursor next to the image.
+            int padding = DisplayUtils.dpToPx(getActivity(), 16) * 2;
+            mFeaturedImageThumbnailWidth -= padding;
+        }
+
+        return mFeaturedImageThumbnailWidth;
+    }
+
+    public void setFeaturedImage(String imageURL) {
+        mFeaturedImageURL = imageURL;
+        if (!mActivity.getSupportActionBar().isShowing()) {
+            mActivity.getSupportActionBar().show();
+        }
+        mActivity.mSectionsPagerAdapter.notifyDataSetChanged();
+    }
+
+    @Override
+    public void updateFragment() {
+        mSetFeaturedImageButton.setVisibility(View.GONE);
+        mRemoveFeaturedImageButton.setVisibility(View.VISIBLE);
+
+        if (mFeaturedImageURL.contains("wordpress.com")) {
```

Comment:
This results in a crash when selecting media from a self-hosted site. Take a look at #2509 for reference.